src/devices/Card/Ultralight/UltralightCard.cs

```diff
@@ -0,0 +1,995 @@
+            configuration.Mirror.Position = (byte)((Data[0] >> 4) & 0b0000_0011);
+            configuration.IsStrongModulation = ((Data[0] >> 2) & 1) == 1;
+            configuration.Mirror.Page = Data[2];
+            configuration.Authentication.AuthenticationPageRequirement = Data[3];
+            configuration.Authentication.IsReadWriteAuthenticationRequired = ((Data[4] >> 7) & 1) == 1;
+            configuration.Authentication.IsWritingLocked = ((Data[4] >> 6) & 1) == 1;
+            configuration.IsSleepEnabled = ((Data[0] >> 3) & 1) == 1;
+            configuration.FieldDetectPin = (FieldDetectPin)(Data[0] & 0b0000_0011);
+            configuration.NfcCounter.IsEnabled = ((Data[4] >> 4) & 1) == 1;
+            configuration.NfcCounter.IsPasswordProtected = ((Data[4] >> 3) & 1) == 1;
+            configuration.Authentication.MaximumNumberOfPossibleTries = (byte)(Data[4] & 0b0000_0111);
+            return true;
+        }
+
+        /// <summary>
+        /// Write the configuration
+        /// </summary>
+        /// <param name="configuration">The configuration to write</param>
+        /// <returns>True if success</returns>
+        /// <remarks>An authentication has to happen and will use the credentials stored</remarks>
+        public bool WriteConfiguration(Configuration configuration)
+        {
+            if (UltralightCardType is UltralightCardType.UltralightC)
+            {
+                // There is no configuration for this card
+                return false;
+            }
+
+            var res = ProcessAuthentication(AuthenticationKey);
+            if (!res)
+            {
+                return false;
+            }
+
+            var serialized = configuration.Serialize();
+            BlockNumber = (byte)(NumberBlocks - 4);
+            if (UltralightCardType is UltralightCardType.UltralightNtagI2cNT3H1101 or UltralightCardType.UltralightNtagI2cNT3H1101W0 or UltralightCardType.UltralightNtagI2cNT3H2111W0
+                or UltralightCardType.UltralightNtagI2cNT3H2101 or UltralightCardType.UltralightNtagI2cNT3H1201W0 or UltralightCardType.UltralightNtagI2cNT3H2211W0)
+            {
+                // Those are using different configurations, Configuration registers and session registers
+                BlockNumber--;
+            }
+
+            Data = new byte[4];
+            serialized.AsSpan(0, 4).CopyTo(Data);
+            Command = UltralightCommand.Write4Bytes;
+            var ret = RunUltralightCommand();
+            if (ret < 0)
+            {
+                return false;
+            }
+
+            BlockNumber++;
+            serialized.AsSpan(4, 4).CopyTo(Data);
+            Command = UltralightCommand.Write4Bytes;
+            ret = RunUltralightCommand();
+            if (ret < 0)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Set the password, the AuthenticationKey is used as the old password
```

Comment:
would be useful to type what's the expected size of the buffer

Reply:
it's done in the ProcessAuthentication which is the function called from there. So the buffer size is checked.